tokio/src/stream/timeout.rs

```diff
@@ -0,0 +1,63 @@
+use crate::stream::{Fuse, Stream};
+use crate::time::{Delay, Elapsed, Instant};
+
+use core::future::Future;
+use core::pin::Pin;
+use core::task::{Context, Poll};
+use pin_project_lite::pin_project;
+use std::time::Duration;
+
+pin_project! {
+    /// Stream returned by the [`timeout`](super::StreamExt::timeout) method.
+    #[must_use = "streams do nothing unless polled"]
+    #[derive(Debug)]
+    pub struct Timeout<S> {
+        #[pin]
+        stream: Fuse<S>,
+        #[pin]
```

Comment:
I'm surprised this needs to be `#[pin]` as I *believe* the type itself is `Unpin`, but it doesn't really matter either way :) (just musing).

Reply:
I guess it doesn't have to be. Removed the #[pin]